pdfminer/pdfdocument.py

```diff
@@ -500,28 +500,74 @@ def get_cfm(self, name):
 
     def authenticate(self, password):
         password = password.encode('utf-8')[:127]
```

Comment:
The password needs to be passed through SASLprep for Unicode normalisation. If not, the authentication procedure could produce incorrect output.

Reply:
See [here](https://github.com/pdf-association/safedocs/tree/main/Unicode%20passwords) for relevant test docs.